src/go/rdctl/pkg/reg/reg.go

```diff
@@ -70,32 +93,34 @@ func convertToRegFormat(pathParts []string, v reflect.Value, jsonTag string) ([]
 		retLines = append(retLines, nestedReturnedLines...)
 		return retLines, nil
 	case reflect.Ptr:
-		if v.IsNil() {
-			return nil, nil
-		} else {
-			return convertToRegFormat(pathParts, v.Elem(), jsonTag)
-		}
+		return convertToRegFormat(pathParts, structType.Elem(), value, jsonTag, path)
 	case reflect.Slice, reflect.Array:
-		// Currently, all arrays in the options are arrays of strings
-		numValues := v.Len()
-		if numValues == 0 {
-			return nil, nil
+		if value.Kind() != reflect.Slice && value.Kind() != reflect.Array {
+			return nil, fmt.Errorf("expected slice or array at %s, got %v", path, value.Kind())
 		}
+		// Currently, all arrays in the options are arrays of strings
+		numValues := value.Len()
 		arrayValues := make([]string, numValues)
 		for i := 0; i < numValues; i++ {
-			arrayValues[i] = v.Index(i).String()
+			item := value.Index(i)
+			for item.Kind() == reflect.Interface || item.Kind() == reflect.Pointer {
+				item = item.Elem()
+			}
+			arrayValues[i] = item.String()
 		}
 		return []string{fmt.Sprintf(`"%s"=hex(7):%s`, jsonTag, stringToMultiStringHexBytes(arrayValues))}, nil
 	case reflect.Map:
-		numValues := len(v.MapKeys())
-		if numValues == 0 {
-			return nil, nil
-		}
+		// If it's a map (always of string => bool|string|int), the typed and actual values are the same
+		// The only difference is that if the field isn't specified in the input, there will be an instance
+		// in `structType` but not `value`.
 		returnedLines := []string{fmt.Sprintf("[%s]", strings.Join(pathParts, "\\"))}
-		typedKeys := utils.SortKeys(v.MapKeys())
-		for _, typedKey := range typedKeys {
-			keyAsString := typedKey.StringKey
-			innerLines, err := convertToRegFormat(append(pathParts, keyAsString), v.MapIndex(typedKey.MapKey), keyAsString)
+		actualKeys := utils.SortKeys(value.MapKeys())
+		for _, actualKey := range actualKeys {
+			keyAsString := actualKey.StringKey
+			// If it's a map (always of string => bool|string|int), the typed and actual values are the same
+			// The only difference is that if the field isn't specified in the input, there will be an instance
+			// in `typedValue` but not `actualValue`.
```

Comment:
This comment is repeated a few lines up. (Also, there's no "typed value".)

Reply:
And with the same rename as in plutil.go, both comments can be dropped.